src/lib/handlers/ack.ts

```diff
@@ -117,19 +117,26 @@ const handleAck: PacketHandler = (client, packet) => {
 			client.messageIdProvider.deallocate(messageId)
 			const granted = packet.granted as number[]
 			for (let grantedI = 0; grantedI < granted.length; grantedI++) {
-				if ((granted[grantedI] & 0x80) !== 0) {
+				const subackRC = granted[grantedI]
+				if ((subackRC & 0x80) !== 0) {
+					err = new Error(`Subscribe error: ${ReasonCodes[subackRC]}`)
+					err.code = subackRC
+
 					// suback with Failure status
 					const topics = client.messageIdToTopic[messageId]
 					if (topics) {
 						topics.forEach((topic) => {
 							delete client['_resubscribeTopics'][topic]
 						})
 					}
+					client['_removeOutgoingAndStoreMessage'](messageId, () => {
+						cb(err, packet)
+					})
```

Comment:
In this way cb is called twice once here and once later

Reply:
removed